test/browser/render.js

```diff
@@ -588,44 +699,69 @@ describe('render()', () => {
 		expect(sortAttributes(html)).to.equal(sortAttributes('<input type="range" min="0" max="100" list="steplist">'));
 	});
 
-	it('should not execute append operation when child is at last', (done) => {
+	it('should not execute append operation when child is at last', () => {
+		// See developit/preact#717 for discussion about the issue this addresses
+
+		let todoText = 'new todo that I should complete';
 		let input;
+		let setText;
+		let addTodo;
+
+		const ENTER = 13;
+
 		class TodoList extends Component {
 			constructor(props) {
 				super(props);
 				this.state = { todos: [], text: '' };
-				this.setText = this.setText.bind(this);
-				this.addTodo = this.addTodo.bind(this);
+				setText = this.setText = this.setText.bind(this);
+				addTodo = this.addTodo = this.addTodo.bind(this);
 			}
 			setText(e) {
 				this.setState({ text: e.target.value });
 			}
-			addTodo() {
-				let { todos, text } = this.state;
-				todos = todos.concat({ text });
-				this.setState({ todos, text: '' });
+			addTodo(e) {
+				if (e.keyCode === ENTER) {
+					let { todos, text } = this.state;
+					todos = todos.concat({ text });
+					this.setState({ todos, text: '' });
+				}
 			}
 			render() {
-				const {todos, text} = this.state;
+				const { todos, text } = this.state;
 				return (
-					<div onKeyDown={ this.addTodo }>
-						{ todos.map( todo => (<div>{todo.text}</div> )) }
+					<div onKeyDown={this.addTodo}>
+						{ todos.map( todo => ([
+							<span>{todo.text}</span>,
```

Comment:
The original issue called out this example as a better test for this test case so I updated the test to cover this.